altair/utils/data.py

```diff
@@ -398,18 +398,12 @@ def _data_to_csv_string(data: dict | pd.DataFrame | DataFrameLike) -> str:
             msg = "pandas is required to convert a dict to a CSV string"
             raise ImportError(msg) from exc
         return pd.DataFrame.from_dict(data["values"]).to_csv(index=False)
-    elif isinstance(data, DataFrameLike):
-        # experimental interchange dataframe support
-        import pyarrow as pa
-        import pyarrow.csv as pa_csv
-
-        pa_table = arrow_table_from_dfi_dataframe(data)
-        csv_buffer = pa.BufferOutputStream()
-        pa_csv.write_csv(pa_table, csv_buffer)
-        return csv_buffer.getvalue().to_pybytes().decode()
-    else:
-        msg = "to_csv only works with data expressed as " "a DataFrame or as a dict"
-        raise NotImplementedError(msg)
+    try:
+        data_nw = nw.from_native(data, eager_only=True)
+    except TypeError as exc:
+        msg = "to_csv only works with data expressed as a DataFrame or as a dict"
+        raise NotImplementedError(msg) from exc
+    return data_nw.write_csv()
```

Comment:
in order to be able to use `write_csv` in Narwhals, `v1.5.2+` is required - is that OK?

I'm not aware of anyone putting upper bounds on Narwhals (in theory that shouldn't ever be necessary) so this shouldn't cause issues

Reply:
> in order to be able to use `write_csv` in Narwhals, `v1.5.2+` is required - is that OK?

Yeah seems fine to me.
Upgrading (my env) silenced this warning from `pylance` for me: 

```
> Cannot access attribute "write_csv" for class "narwhals.stable.v1.DataFrame[NativeFrame | narwhals.dataframe.DataFrame[Any] | narwhals.typing.DataFrameLike | altair.utils.core.DataFrameLike]"
> > Attribute "write_csv" is unknown

```